scrapyd_k8s/object_storage/libcloud_driver.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+from libcloud.storage.types import ObjectError, ContainerDoesNotExistError, ObjectDoesNotExistError, InvalidContainerNameError
+from libcloud.storage.providers import get_driver
+
+class LibcloudObjectStorage:
+
+    def __init__(self, config):
+        self._region = config.scrapyd().get('aws_region')
+        self._access_key_id = config.scrapyd().get('aws_access_key_id')
+        self._secret_access_key = config.scrapyd().get('aws_secret_access_key')
+        self._container_name = config.scrapyd().get('aws_bucket_name')
+        self._storage_provider = config.scrapyd().get('storage_provider')
+
+        if self._region is None or self._access_key_id is None or self._secret_access_key is None:
+            raise ValueError('AWS credentials not set')
+        if self._container_name is None:
+            raise ValueError('Bucket name not set')
+
+        driver_class = get_driver(self._storage_provider)
+        self.driver = driver_class(self._access_key_id, self._secret_access_key, region=self._region)
```

Comment:
Ideally, we'd have some kind of config that is passed to libcloud, so we don't need to put all known configuration keys for libcloud here. I'll look into it when I have some time.